apps/system/js/app_chrome.js

```diff
@@ -340,6 +373,67 @@
     window.dispatchEvent(new CustomEvent('global-search-request'));
   };
 
+  AppChrome.prototype.onClickSiteIcon = function onClickSiteIcon() {
+    this.pinningPref && this.pinningPref.get(function(enabled) {
+      if (!enabled || !this.app.isBrowser()) {
+        this.app.debug('Pinning the web disabled');
+        return;
+      }
+      this.setPinDialogCard();
+    }.bind(this));
+  };
+
+  AppChrome.prototype.setPinDialogCard = function ac_setPinDialogCard(url) {
+    var card = new GaiaPinCard();
+    card.title = this.title.textContent;
+    card.icon = this.siteIcon.style.backgroundImage;
+    this.pinCardContainer.innerHTML = '';
+    this.app.getScreenshot(function() {
+      card.background = {
+        src: URL.createObjectURL(this.app._screenshotBlob),
+        themeColor: this.app.themeColor
+      };
+    }.bind(this));
+    this.pinCardContainer.appendChild(card);
+    this.setOrigin();
+    this.pinDialog.classList.remove('hidden');
+  };
+
+  AppChrome.prototype.setOrigin = function ac_setOrigin() {
```

Comment:
Yeah, we already talked about this one. Let's use publicsuffix.org in a separate bug. Thanks!